Automattic/pocket-casts-android
# Add bookmarks from the player shelf

## Description

You can now add bookmarks from the full screen player.

It uses another activity so I could add `android:windowSoftInputMode="adjustResize"` and have the "Save bookmark" button just above the keyboard. I'm not really happy with the amount of code to setup the activity but it might be worth it to be able to add a bookmark quickly. 

## Testing Instructions
1. Play an episode
2. Tap on the mini player to open the full screen player
3. Tap on the shelf three dots icon
4. Tap Add bookmark
5. ✅ Verify the "Add bookmark" page is shown
6. Tap "Save bookmark"
7. ✅ Verify a Snackbar is shown
8. Tap the "View" action in the Snackbar
9. ✅ Verify it takes you to the bookmarks tab and the new bookmark is present
10. Tap the "Now Playing" 
11. Tap on the shelf three dots icon
12. Tap the edit pencil
13. Use the drag handles to move the "Add bookmark" item up to the "Shortcut on player"
14. Tap the back button
15. ✅ Verify the "Add bookmark" icon is on the shelf
16. Tap the icon
17. ✅ Verify it opens the add bookmark page

## Screenshots or Screencast 
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/c593d6d7-9ed1-41a5-ae60-b5341d67ba8a" />
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/41c1552a-5901-4966-ab07-7940cea0084c" />
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/af742050-b764-456f-bfeb-63ffbf5efe44" />
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/471f38d9-9e8b-473a-a30f-8ef4aa7da710" />

https://github.com/Automattic/pocket-casts-android/assets/308331/1786c8dd-42d5-4642-b6aa-3c383cac2ed0
